src/components/ProfilePage/PersonalDetails.tsx

```diff
@@ -0,0 +1,131 @@
+              Cancel
+            </Button>
+            {loading ? (
+              <Button className="flex gap-2">
+                <ReloadIcon className="animate-spin" />
+                Saving...
+              </Button>
+            ) : (
+              <Button onClick={onSave}>Save</Button>
+            )}
+          </div>
+        )}
+      </div>
+      <div className="p-5 flex flex-col gap-3">
+        <div>
+          {editable ? (
+            <>
+              <label htmlFor="user_name" className="text-sm">
+                Username
+              </label>
+              <Input
+                id="user_name"
+                name="name"
+                value={username}
+                onChange={(e) => setUsername(e.target.value)}
+              />
+            </>
+          ) : (
+            <div className="flex items-center gap-5">
+              <p>Username: </p>
+              <p>{user?.username}</p>
+            </div>
+          )}
+        </div>
+
+        <div className="flex items-center justify-center gap-5">
+          <div className="flex-1">
+            {editable ? (
+              <>
+                <label htmlFor="user_first" className="text-sm">
+                  Legal Name
+                </label>
+                <div className="flex items-center gap-5">
+                  <Input
+                    id="user_first"
+                    name="first_name"
+                    value={firstName}
+                    onChange={(e) => setFirstName(e.target.value)}
+                  />
+                  <Input
+                    id="user_last"
+                    name="last_name"
+                    value={lastName}
+                    onChange={(e) => setLastName(e.target.value)}
+                  />
+                </div>
+              </>
+            ) : (
+              <div className="flex items-center gap-5">
+                <p>Legal Name: </p>
+                <p>{`${user?.firstName} ${user?.lastName}`}</p>
+              </div>
+            )}
+          </div>
+        </div>
+        <div className="flex items-center justify-center gap-5">
+          <div className="flex-1">
+            {editable ? (
+              <>
+                <label htmlFor="nick_name" className="text-sm">
+                  Artist Nick name
+                </label>
+                <div className="flex items-center gap-5">
+                  <Input
+                    id="nick_name"
+                    name="nick_name"
+                    value={nickName}
+                    onChange={(e) => setNickName(e.target.value)}
+                  />
+                </div>
+              </>
+            ) : (
+              <div className="flex items-center gap-5">
+                <p>Artist Nick name: </p>
+                <p>{user?.nickName}</p>
+              </div>
+            )}
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default PersonalDetails;
```

Comment:
this component is too big
- rule: all files should have less than `100 lines of code`
- actual: this file has `131 lines of code`

some components you could split out:
- `<ArtistName />`
- `<LegalName />`
- `<Username />`
- `<ProfileButtons />`
- etc